ahcore/readers.py

```diff
@@ -295,6 +292,74 @@ def read_region(self, location: tuple[int, int], level: int, size: tuple[int, in
 
         return pyvips.Image.new_from_array(stitched_image.transpose(1, 2, 0))
 
+    def _read_feature_region(self, size: tuple[int, int], location: tuple[int, int]) -> pyvips.Image:
+        assert self._num_samples is not None
+
+        image_dataset: h5py.Dataset = self._file["data"]
+
+        assert image_dataset is not None
+
+        x, y = location
+        w, h = size
+
+        if self._stitching_mode != StitchingMode.CROP:
+            raise NotImplementedError("Stitching mode other than CROP is not supported for features.")
+
+        if image_dataset.shape[0] != self._num_samples:
+            raise ValueError(
+                f"Reading features expects that the saved feature vectors are the same "
+                f"length as the number of samples in the dataset. "
+                f"Feature vector length was {image_dataset.shape[0]}, "
+                f"number of samples in the dataset was {self._num_samples}"
+            )
+
+        if x + w > self._num_samples or y + h > 1:
+            raise ValueError(
+                f"Feature vectors are saved as (num_samples, 1) "
+                f"and the requested size {size} at location {location} is too large."
+            )
+
+        # this simplified version of the crop is done as it is faster than the crop for images crop
+        return pyvips.Image.new_from_array(np.expand_dims(image_dataset[x : x + w, :], axis=0).astype(np.float32))
+
+    def read_region(self, location: tuple[int, int], level: int, size: tuple[int, int]) -> pyvips.Image:
+        """
+        Reads a region in the stored h5 file. This function stitches the regions as saved in the cache file. Doing this
+        it takes into account:
```

Comment:
Mention somewhere the difference between this function and the feature function (e.g. interpolation etc)

Reply:
Added comments to the function